src/ovro_lwa_portal/io.py

```diff
@@ -65,80 +62,28 @@ def _normalize_doi(source: str) -> str:
-
-    Returns
-    -------
-    str
-        URL to the zarr store.
-
-    Raises
-    ------
-    DataSourceError
-        If DOI cannot be resolved or zarr URL not found.
-    """
+    """Resolve DOI to zarr URL using DataCite Media API."""
     try:
         import requests
     except ImportError as e:
-        msg = (
-            "requests library is required for DOI resolution. "
-            "Install with: pip install requests"
-        )
+        msg = "requests is required for DOI resolution. Install with: pip install requests"
         raise ImportError(msg) from e
 
-    # Use DataCite Media API to get the zarr store URL
-    # Based on: https://github.com/caltechlibrary/caltechdata_api/blob/main/caltechdata_api/download_file.py
     api_url = f"https://api.datacite.org/dois/{doi}/media"
-    logger.info(f"Querying DataCite Media API: {api_url}")
 
     try:
-        response = requests.get(api_url, timeout=10)
-        response.raise_for_status()
-        data = response.json()
-
-        if "data" not in data or not data["data"]:
-            msg = f"No media files found for DOI: {doi}"
-            raise DataSourceError(msg)
-
-        # Look for zarr store in media types
-        # Prefer application/zarr or application/zip+zarr media types
-        zarr_url = None
-        for media in data["data"]:
-            attrs = media.get("attributes", {})
-            media_type = attrs.get("mediaType", "")
-            url = attrs.get("url", "")
-
-            # Check if this is a zarr store
-            if "zarr" in media_type.lower() or url.endswith(".zarr") or ".zarr/" in url:
-                zarr_url = url
-                logger.info(f"Found zarr store with media type '{media_type}': {url}")
-                break
-
-        # If no explicit zarr media type, use the first URL
-        if zarr_url is None and data["data"]:
-            zarr_url = data["data"][0]["attributes"]["url"]
-            logger.warning(f"No zarr media type found, using first URL: {zarr_url}")
-
-        if zarr_url is None:
-            msg = f"No zarr store URL found for DOI: {doi}"
-            raise DataSourceError(msg)
-
-        logger.info(f"Resolved DOI to zarr URL: {zarr_url}")
-        return zarr_url
-
-    except requests.RequestException as e:
-        msg = f"Failed to query DataCite Media API for DOI {doi}: {e}"
-        raise DataSourceError(msg) from e
-    except (KeyError, IndexError) as e:
-        msg = f"Unexpected response format from DataCite Media API for DOI {doi}: {e}"
+        r = requests.get(api_url, timeout=10)
+        r.raise_for_status()
+        data = r.json()["data"]
+
+        # Prefer zarr media type, fallback to first URL
+        for media in data:
+            if media["attributes"]["mediaType"] == "application/zarr":
+                return media["attributes"]["url"]
+
+        return data[0]["attributes"]["url"]
+    except Exception as e:
+        msg = f"Failed to resolve DOI {doi}: {e}"
```

Comment:
Looks like this code was taken from https://github.com/caltechlibrary/caltechdata_api/blob/main/caltechdata_api/download_file.py. Let's use use the caltechdata_api directly and use the code there instead of rewriting here, so that it's maintained upstream.